flang/include/flang/Lower/FIRBuilder.h

```diff
@@ -243,8 +249,12 @@ class FirOpBuilder : public mlir::OpBuilder {
 
   /// Construct one of the two forms of shape op from an array box.
   mlir::Value consShape(mlir::Location loc, const fir::AbstractArrayBox &arr);
+  mlir::Value consShape(mlir::Location loc, llvm::ArrayRef<mlir::Value> shift,
+                        llvm::ArrayRef<mlir::Value> exts);
+  mlir::Value consShape(mlir::Location loc, llvm::ArrayRef<mlir::Value> exts);
 
-  /// Create one of the shape ops given an extended value.
+  /// Create one of the shape ops given an extended value. For a boxed value,
+  /// this may create a `fir.shift` op even when it is not necessary.
```

Comment:
When we have a value of `box` type, that value must have been created with a `fir,shift` sort of op initially. That is, the origin shift information is included as part of the box. Therefore, in some sense, it may not be strictly necessary to pull the shift information out of the box and re-create a new ShiftOp to group that information. In one of the cases I was looking at, IIRC, there was an `embox` that was then used to create a new `shift` op.

I can remove this part of the comment. It's probably not that helpful.